File: 26.12.25/find_index.java
class Solution {
    public int strStr(String haystack, String needle) {
        int n = haystack.length();
        int k  = needle.length();

        for(int i = 0;i<=n-k;i++){
            String subString = haystack.substring(i,i+k); //create the substring 
            if(subString.equals(needle))return i; //compare retuen the result
        }

        return -1;
    }
}

File: 01.01.26/find_index.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        i=0
        if needle in haystack:
            while i<len(haystack):
                for m in range(len(needle)):
                    if haystack[m+i] != needle[m]:
                        i=i+1
                        break
                    elif m==len(needle)-1:
                        return i
        else:
            return -1
        
